File: src/main/scala/org/ranceworks/postclustering/token/JapaneseTokenizer.scala
package org.ranceworks.postclustering.token

import collection.JavaConverters._

class JapaneseTokenizer extends Tokenizer {


  override def tokenize(text: String): List[String] = {
    val tokenizer =  new com.atilika.kuromoji.ipadic.Tokenizer()
    tokenizer.tokenize(text).asScala.toList.map(token => token.getBaseForm)
  }
}

File: src/main/java/org/ranceworks/postclustering/tokenizer/JapaneseTokenizer.java
package org.ranceworks.postclustering.tokenizer;

import com.atilika.kuromoji.ipadic.Token;

import java.io.IOException;
import java.util.List;
import java.util.stream.Collectors;

public class JapaneseTokenizer implements Tokenizer {
     @Override
     public List<String> tokenize(String text) throws IOException {
         com.atilika.kuromoji.ipadic.Tokenizer tokenizer =  new com.atilika.kuromoji.ipadic.Tokenizer();
         return tokenizer.tokenize(text).parallelStream().map(Token::getBaseForm).collect(Collectors.toList());
     }
}
